public/app/plugins/datasource/loki/language_provider.ts

```diff
@@ -325,22 +363,31 @@ export default class LokiLanguageProvider extends LanguageProvider {
     return ['{', cleanSelector, '}'].join('');
   }
 
+  async getSeriesLabels(selector: string, absoluteRange: AbsoluteTimeRange) {
+    if (this.lookupsDisabled) {
+      return undefined;
+    }
+    try {
+      return await this.fetchSeriesLabels(selector, absoluteRange);
+    } catch (error) {
+      // TODO: better error handling
+      console.error(error);
+      return undefined;
+    }
+  }
+
+  /**
+   * Fetches all label keys
+   * @param absoluteRange Fetches
+   */
   async fetchLogLabels(absoluteRange: AbsoluteTimeRange): Promise<any> {
     const url = '/api/prom/label';
     try {
       this.logLabelFetchTs = Date.now();
       const rangeParams = absoluteRange ? rangeToParams(absoluteRange) : {};
       const res = await this.request(url, rangeParams);
-      const labelKeys = res.data.data.slice().sort();
-
-      this.labelKeys = {
-        ...this.labelKeys,
-        [EMPTY_SELECTOR]: labelKeys,
-      };
-      this.labelValues = {
-        [EMPTY_SELECTOR]: {},
-      };
-      this.logLabelOptions = labelKeys.map((key: string) => ({ label: key, value: key, isLeaf: false }));
+      this.labelKeys = res.slice().sort();
```

Comment:
In pure js there's one kinda hacky thing you can do to make slice even faster, but I am not sure if typescript does this optimization on its own. You can pass a begin index to slice.
```ts
this.labelKeys = res.slice(0).sort();
```